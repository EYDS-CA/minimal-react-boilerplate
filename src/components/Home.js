import React from "react";
import { Link } from "react-router-dom";

const Home = props => (
  <div>
    <h1>Basic React App</h1>
    {props.name && <p>Hello, {props.name}!</p>}
    <Link to="/about-us">Go to About Us</Link>
  </div>
);

export default Home;
