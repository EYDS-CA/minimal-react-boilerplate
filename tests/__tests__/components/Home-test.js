import React from "react";
import { shallow } from "enzyme";
import Home from "@/components/Home";

// Example snapshot test for component regression testing
it("renders correctly with no name", () => {
  const component = shallow(<Home />);
  expect(component).toMatchSnapshot();
});

// Example snapshot test with props provided
it("renders correctly with a name", () => {
  const component = shallow(<Home name="Developer" />);
  expect(component).toMatchSnapshot();
});
