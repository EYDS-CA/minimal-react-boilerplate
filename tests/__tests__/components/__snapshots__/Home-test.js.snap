// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with a name 1`] = `
<div>
  <h1>
    Basic React App
  </h1>
  <p>
    Hello, 
    Developer
    !
  </p>
  <Link
    to="/about-us"
  >
    Go to About Us
  </Link>
</div>
`;

exports[`renders correctly with no name 1`] = `
<div>
  <h1>
    Basic React App
  </h1>
  <Link
    to="/about-us"
  >
    Go to About Us
  </Link>
</div>
`;
